src/context/TimerContext.tsx

```diff
@@ -0,0 +1,89 @@
+import { createContext, useContext, useState } from 'react';
+import type { Timer, TimerContextType } from "../types/types";
+
+const TimerContext = createContext<TimerContextType | undefined>(undefined);
+
+export const TimerProvider = ({ children }: { children: any }) => {
+    const [timers, setTimers] = useState<Timer[]>([]);
+    const [currentTimerIndex, setCurrentTimerIndex] = useState(0);
+    const [isWorkoutRunning, setIsWorkoutRunning] = useState(false);
+
+    const addTimer = (timer: Timer) => {
+        setTimers(prevTimers => [...prevTimers, timer]);
+    };
+
+    const removeTimer = (id: string) => {
+        const updatedTimers = timers.filter(timer => timer.id !== id);
+        setTimers(updatedTimers);
+        if (currentTimerIndex >= updatedTimers.length) {
+            setCurrentTimerIndex(Math.max(0, currentTimerIndex - 1));
+        }
+    };
+    const startWorkout = () => {
+        setIsWorkoutRunning(true);
+        if (timers.length > 0) {
+            setTimers(prevTimers => prevTimers.map((timer, index) =>
+                (index === currentTimerIndex ? { ...timer, state: 'running' } : timer)
+            ));
+        }
+    };
+
+    const pauseWorkout = () => {
+        setIsWorkoutRunning(false);
+        setTimers(prevTimers => prevTimers.map(timer => (timer.state === 'running' ? { ...timer, state: 'notRunning' } : timer)));
+    };
+
+    const resetWorkout = () => {
+        setCurrentTimerIndex(0);
+        setIsWorkoutRunning(false);
+        setTimers(prevTimers =>
+            prevTimers.map(timer => ({
+                ...timer,
+                state: 'notRunning',
+                timeLeft: timer.type === 'stopwatch' ? 0 : timer.config.initialTime || 0,
+                currentRound: timer.type === 'xy' ? timer.config.numberOfRounds : undefined
+            }))
+        );
+    };
+
+
+    const fastForward = () => {
```

Comment:
This is not working correctly. 

1) When I fast-forward, I expect the timer to reach it's end state and move to the next timer. Instead, the active timer switches to the next timer (correct), but the previous timer is still counting down (incorrect).

2) Let's say I have 3 timers: 
- Timer A: 10 seconds
- Timer B: 2 minutes
- Timer C: 5 Seconds
I start my workout, wait 5 seconds, and then click fast-forward during Timer A. Timer B then becomes active, but Timer A is still counting. Once Timer A is complete (5 seconds later), it sends me to my next timer, Timer C, even though I still have a lot of time left of Timer B.